### PR TITLE
Add support for GCPOLL_INLINE for crossgen

### LINKED ISSUE #13591: CrossGen can't support GCPOLL_INLINE
During the [suppress GC transition](https://github.com/dotnet/coreclr/pull/26458) work, it was discovered that CrossGen can't support indirect memory addresses that are required for `GCPOLL_INLINE`. This work was discussed with @davidwrighton and attempted, but was blocked by issues referenced in https://github.com/dotnet/coreclr/issues/27186.

/cc @davidwrighton @MichalStrehovsky @sergiy-k

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #13591

### CODE REVIEW

**Comment by jkotas** [comment]
cc @dotnet/crossgen-contrib

File: src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs [hunk 1]
```diff
@@ -2443,12 +2443,6 @@ private uint getThreadTLSIndex(ref void* ppIndirection)
         private void* getInlinedCallFrameVptr(ref void* ppIndirection)
         { throw new NotImplementedException("getInlinedCallFrameVptr"); }
 
-        private int* getAddrOfCaptureThreadGlobal(ref void* ppIndirection)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs:2446]: What about CoreRT? Is this something that we will need to handle during that merge?

Comment (jkotas) [reply src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs:2446]: Yes, it is the usual procedure.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by erozenfeld — APPROVED** [review]
LGTM

**Review by davidwrighton — APPROVED** [review]

**Comment by jkotas** [comment]
Test failures are #39376 and #24444

**Pull request merged**